Source/CombatExtended/Compatibility/TurretRegistry.cs

```diff
@@ -0,0 +1,185 @@
+	    getGun       = new Dictionary<Type, GetGunFunction>();
+	    
+        }
+
+        public static void RegisterReloadableTurret(Type turretType, SetReloadingFunction setReload, GetReloadingFunction getReload, GetGunFunction gun, GetAmmoFunction ammo=null)
+        {
+            if (!enabled)
+            {
+                Enable();
+            }
+	    getReloading[turretType] = getReload;
+	    setReloading[turretType] = setReload;
+
+	    getAmmo[turretType] = ammo;
+	    getGun[turretType] = gun;
+        }
+
+	public static void SetReloading(this Building_Turret turret, bool reloading) {
+	    var ceturret = turret as Building_TurretGunCE;
+	    if (ceturret!=null) {
+		ceturret.isReloading = reloading;
+		return;
+	    }
+	    if (enabled) {
+		SetReloadingFunction func;
+		if (setReloading.TryGetValue(turret.GetType(), out func)) {
+		    func(turret, reloading);
+		    return;
+		}
+	    }
+	    CELogger.Warn("Asked to set reloading on an unknown turret type: "+turret);
+	}
+
+	public static bool GetReloading(this Building_Turret turret) {
+	    var ceturret = turret as Building_TurretGunCE;
+	    if (ceturret!=null) {
+		return ceturret.isReloading;
+	    }
+	    if (enabled) {
+		GetReloadingFunction func;
+		if (getReloading.TryGetValue(turret.GetType(), out func)) {
+		    return func(turret);
+		}
+	    }
+	    CELogger.Warn("Asked to get reloading on an unknown turret type: "+turret);
+	    return false;
+	}
+
+	public static CompAmmoUser GetAmmo(this Building_Turret turret) {
+	    var ceturret = turret as Building_TurretGunCE;
+	    if (ceturret!=null) {
+		return ceturret.CompAmmo;
+	    }
+	    if (enabled) {
+		GetAmmoFunction func;
+		if (getAmmo.TryGetValue(turret.GetType(), out func)) {
+		    return func(turret);
+		}
+		GetGunFunction gfunc;
+		if (getGun.TryGetValue(turret.GetType(), out gfunc)) {
+		    return gfunc(turret)?.TryGetComp<CompAmmoUser>();
+		}
+		
+	    }
+	    CELogger.Warn("Asked to get ammo on an unknown turret type: "+turret);
+	    return null;
+	}
+
+	public static Thing GetGun(this Building_Turret turret) {
+	    var ceturret = turret as Building_TurretGunCE;
+	    if (ceturret!=null) {
+		return ceturret.Gun;
+	    }
+	    if (enabled) {
+		GetGunFunction gfunc;
+		if (getGun.TryGetValue(turret.GetType(), out gfunc)) {
+		    return gfunc(turret);
+		}
+		
+	    }
+	    CELogger.Warn("Asked to get gun on an unknown turret type: "+turret);
+	    return null;
+	}
+
+	public static CompMannable GetMannable(this Building_Turret turret) {
+	    var ceturret = turret as Building_TurretGunCE;
+	    if (ceturret!=null) {
+		return ceturret.MannableComp;
+	    }
+	    return turret.TryGetComp<CompMannable>();
+	}
+
+	public static bool GetReloadable(this Building_Turret turret) {
+	    return turret.GetAmmo()?.HasMagazine ?? false;
+	}
+	
+	public static bool ShouldReload(this Building_Turret turret, float threshold=0.5f, bool ensure_ammo_type=true) {
```

Comment:
Is snake case used for default parameters ? I do not recall seeing this in other parts of the code

Reply:
No, that's just me being sloppy after being away fro mthe codebase for too long.

Reply:
Also, that parameter isn't used...